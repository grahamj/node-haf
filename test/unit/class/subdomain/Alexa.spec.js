const sinon = require('sinon');
const { expect } = require('chai');
const proxyquire = require('proxyquire')
  .noPreserveCache().noCallThru();

describe('Alexa class', () => {
  let Alexa;
  let callServiceStub;

  beforeEach(() => {
    callServiceStub = sinon.stub().resolves('success');
    class MediaPlayer {}
    MediaPlayer.prototype.callService = callServiceStub;
    Alexa = proxyquire('../../../../src/class/subdomain/Alexa', {
      '../domain/MediaPlayer.js': MediaPlayer,
    });
  });

  describe('say()', () => {

    it('Calls service with proper args', () => {
      const alexa = new Alexa();
      alexa.entityId = 'fancy entity';
      alexa.say('Hi!');
      expect(callServiceStub.firstCall.args).to.matchSnapshot();
    });

  });

});
