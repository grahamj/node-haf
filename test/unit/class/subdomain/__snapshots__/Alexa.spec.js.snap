// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alexa class say() Calls service with proper args 1`] = `
Array [
  Object {
    "domain": "notify",
    "service": "alexa_media",
    "service_data": Object {
      "message": "Hi!",
      "target": "fancy entity",
    },
  },
]
`;
